libmamba/src/api/create.cpp

```diff
@@ -30,13 +33,98 @@ namespace mamba
         config.load();
 
         auto& create_specs = config.at("specs").value<std::vector<std::string>>();
+        auto& clone_env_name = config.at("clone_env").value<std::string>();
         auto& use_explicit = config.at("explicit_install").value<bool>();
         auto& json_format = config.at("json").get_cli_config<bool>();
         auto& env_vars = config.at("spec_file_env_vars").value<std::map<std::string, std::string>>();
         auto& no_env = config.at("no_env").value<bool>();
 
         auto channel_context = ChannelContext::make_conda_compatible(ctx);
 
+        // Handle clone environment logic
+        std::vector<std::string> clone_specs;
+        if (!clone_env_name.empty())
```

Comment:
Could you move this code into a dedicated function so that we can write something like
```cpp
auto clone_specs = get_clone_specs(/* parameters needed */);
```

The idea is to have this function "short" and easy to read (it would require some further refactoring, but this is orthogonal to your PR).